docs/setup/operation/upgrading.md

```diff
@@ -35,6 +35,12 @@ So, copying `notebook` and `conf` directory should be enough.
 
 ## Migration Guide
 
+### Upgrading from Zeppelin 0.8 to 0.9
+
+ - From 0.9, we change the notes file name structure ([ZEPPELIN-2619](https://issues.apache.org/jira/browse/ZEPPELIN-2619)). So when you upgrading zeppelin to 0.9, you need to upgrade note file. Here's steps you need to follow:
```

Comment:
is there a way user can tell they have the "old format"?

Reply:
I am afraid not, the only way to via zeppelin version. Before 0.9.0, it is old format.

Reply:
hmm, if someone is sharing or source control notebook, it will be very hard to tell what's going on...

Reply:
hmm, good point, it would be an issue if user is using git. Let me think about it.

Reply:
@felixcheung I have fixed by adding commit message during note message.
Here's the what I see when I upgrade the built-in zeppelin tutorial notes. But there's no commit message when deleting note. This is a bug of `GitNotebookRepo`

```
commit 7b1e12fe720fcaee916b7be7494e587d766703cd (HEAD -> master)
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Tue Mar 12 15:36:46 2019 +0800

    Upgrade note 'Untitled Note 1' from old format to new format

commit db9de85f65896f02f8ed112d84ca0ba89d09625b
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Tue Mar 12 15:36:46 2019 +0800

    Upgrade note 'Basic Features (Spark)' from old format to new format

commit cf0c44f3c6ecf17ed846ce953c4d281a9623b4b5
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Tue Mar 12 15:36:45 2019 +0800

    Upgrade note 'Using Pig for querying data' from old format to new format

commit b4fb0f512b0f5862983a308c792ea1ab6bd2351f
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Tue Mar 12 15:36:45 2019 +0800

    Upgrade note 'Matplotlib (Python • PySpark)' from old format to new format

commit 6f1d7608adb793f5efa58b9916eb5ff36002f236
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Tue Mar 12 15:36:45 2019 +0800

    Upgrade note 'R (SparkR)' from old format to new format

```

Reply:
ping @felixcheung

Reply:
hm, I'm confused still - git source control is one, but it sounds like there is no way other than trying to open it to know what format the note is saved in. I kinda agree maybe we need a version number or change file extension or something.

for instance, if my team started using zeppelin a year ago, and have long existing notes, and someone upgrade to zeppelin 0.9 or get some note from the internet, would I know what's going on if I open the new note on zeppelin 0.8? or old note on zeppelin 0.9?

Reply:
I have added the version in note file and also update the commit message to include version.
```
commit b5ed201cab6cab13eed59cc1330b29937ae55889 (HEAD -> master)
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Thu Mar 14 15:03:58 2019 +0800

    Upgrade note 'Untitled Note 1' to 0.9.0-SNAPSHOT

commit f4d08fe5043eb6e905dea5ade45b6ccd2458e8c1
Author: Jeff Zhang <zjffdu@apache.org>
Date:   Thu Mar 14 15:03:58 2019 +0800

    Upgrade note 'Basic Features (Spark)' to 0.9.0-SNAPSHOT
```

Reply:
Regarding note compatibility across versions, user can still import note from 0.8 to 0.9 and vice versa. But if user copy 0.8 notes to 0.9 zeppelin instance, zeppelin won't recognize these notes. Because we changed the note file extension and user need to use this upgrade tool to do note upgrade

Reply:
ok